spectrapepper/functions.py

```diff
@@ -2806,3 +2806,45 @@ def asymmetry(data, peak, axis, res= 0.001, s=5, limit=10):
         diff_abs = np.sqrt(diff_abs)*(-1)
     
     return diff_abs
+
+def running_median_nd(data, window_size,plot=False):
+    """
+    Computes the median of an array in a running window defined by the user.
+    The window size needs to be a 2D tuple or list, with the first element being the
+    length of the spectra and the second the total width that will be taken into account
+    for the statistics.
+
+    :type data: numpy array
+    :param data: The spectras 
+
+    :type window_size: tuple/list
+    :param window_size: Window size parameters
+
+    :type plot: boolean
+    :param plot: If you want to plot the new spectra change to True
+
+    :returns: Array containing the computed 1D spectra.
+    :rtype: numpy array[float]
+    """
+    # Check dimensions
+    if np.shape(window_size)[0] != 2:
+        raise ValueError("The window must be 2D")
+    if (window_size[1] % 2) == 0:
+        raise ValueError("The width must be an odd number")
+    # First pad the end and the beginning by duplicating the first and last columns from the spectras
+    pad_length = int((window_size[1]-1)/2)
+    data_new = np.c_[data,data[:,-pad_length:]]
+    data_new2 = np.c_[data[:,:pad_length],data_new]
+    
+    # Get the windows
+    window = np.lib.stride_tricks.sliding_window_view(data_new2,window_size)
+    
+    # Create new spectra from the median of each window
+    new_spectra = []
+    for i in window[0]:
+        new_spectra.append(np.median(i))
+    new_spectra = np.asarray(new_spectra)
```

Comment:
```suggestion
    new_spectra = np.median(window, axis=-1)
```

No need to append to a list then reconvert back to ndarray, numpy can do that much for you:


see the example (for mean) at the bottom of the sliding_window_view docs:
https://numpy.org/doc/stable/reference/generated/numpy.lib.stride_tricks.sliding_window_view.html?highlight=moving%20average

Reply:
Thanks for pointing that out, I did not notice it when I was reading the docs